### ISSUE TITLE
To-Do checklist behavior is unintuitive

### ISSUE THREAD

**Issue by SabreCat** [HabitRPG/habitica#2405]
I was taken by surprise when checking off a To-Do with not all the checklist items completed gave me the full multiplier as if I'd done all of them. I'm not sure the current behavior is intuitive or likely to apply in the majority of use cases.

Here are several ways one might use a checklist for a To-Do in Habit:
1. The To-Do represents a project with a number of smaller actions, all of which must be completed before the project is complete.
2. The To-Do represents a project with a number of smaller actions associated, some of which are optional to complete.
3. The To-Do represents a number of related but independent tasks, grouped together for convenience/compactness.

Only in case 1 does it make sense for a user to receive the full multiplier when checking off the master To-Do: there we can assume that if they're checking off the main task, they _must_ have done everything underneath it. But in the other two cases, we can assume nothing of the sort.

If we changed the logic so that the multiplier is based on the number of _completed_ checklist items, all use cases would be satisfied and the behavior would be intuitive. The situation where it would trip someone up is if they assume case 1 always applies, and use the checklist for listing their actions but aren't diligent about checking them off individually. My instinct is that's not a huge deal--Habit enthusiasts take a lot of satisfaction in checking things off, and the checklist badge lighting up green is a minor reward in itself that people will like to get before completing the To-Do. So the natural course of action will lead people to use the feature appropriately, and if they don't, the feedback of a weaker multiplier will encourage them to make sure to check everything off next time. As it is, users employing use cases 2 or 3 need to adopt the clumsier workflow (relative to style 1 users being made to check everything off) of editing the task to delete all incomplete checklist items before checking the master To-Do.

So, my vote: change To-Do checklists to award bonuses only for completed items. Thoughts?

**Comment by lorian** [comment]
Oh, wow, I actually thought it already worked that way. I'd been occasionally checking off checklists with incomplete items, because they weren't all done but the overlying project was "good enough". I would definitely vote for to-do checklists only giving bonuses for checked-off items. It's a lot less tedious to check off all the items than it is to manually delete uncompleted checklist items!

**Comment by Aiseant** [comment]
For me, it doesn't make sense at all that checking a todo without completing the checklist reward you with the same exp than checking the whole checklist, or even checking some of the items in the list.

If I read the code right, what you propose is already in for checklist in dailies. When the Task is a ToDo, you're granted the exp depending on the checklist length, and nothing else, @lefnire wrote a special case for that. Maybe there was some kind of reason we don't get right now ?

(and by the way, few tests with those and it's clear that there's a big problem with the amount of XP and Gold awarded when checking task and the one you have to give back when you uncheck the task. It's generally a few 1/10 with normal task, but I just saw more than 5xp with checklists ... )

**Comment by wc8** [comment]
@Aiseant (Your _by-the-way_ is a known issue, there are several tickets about it.) 

@SabreCat I agree with _award bonuses only for completed checklist items_
I do not think that _bonuses_ should be awarded for incomplete checklist items.

**Comment by wc8** [comment]
Follow-up thought: I’d prefer over-all the assumption that a ✓ in the box means _good enough_ than _everything done_. 

I don’t want the system to check things off for me. (If it does, at least make it visual, check every box.)

**Comment by patena** [comment]
Why is this closed if it hasn't been either fixed or determined to be the intended behavior? This is confusing. (And now new issues are popping up on the same topic, because I doubt people search closed issues before making new ones...)

(Incidentally, I agree that having the whole ToDo checked should be "good enough", not "everything complete". The current behaviour is unintuitive to me, since it means the checkboxes don't actually do anything at all. In fact I assumed unchecked boxes didn't yield extra xp until I saw an issue about it - I guess I don't look at my numbers closely enough!)

**Comment by SabreCat** [comment]
Well, the behavior _was_ changed: To-Do checklists no longer give you Experience and Gold multiplication for unchecked boxes. They give a multiplier equal to 1 + the number of checked boxes. Per the other issue ticket, though, there are some other questions to be resolved.

**Comment by SabreCat** [comment]
Derp. 1 + _checked_ boxes, if you happened to get the previous error in an email notif xD

**Comment by patena** [comment]
Ohhh! Somehow I entirely missed that the new issue only applies to MP, not to experience and gold. That makes the entire discussion there a bit off topic... I get the feeling that some of the other people there missed that as well. Wow, I'm really not doing well with reading comprehension today. XD
